src/routes/stats/StatsView.vue

```diff
@@ -173,21 +173,35 @@ export default {
     },
   },
   created() {
-    io.socket.get('/stats', (res) => {
+    io.socket.get('/stats/current', (res) => {
       this.loadingData = false;
       if (!res?.length) {
         this.error = true;
         return;
       }
       this.seasons = res;
-      const seasonId = parseInt(this.$route.params.seasonId);
-      this.checkAndSelectSeason(seasonId);
+      [this.selectedSeason] = this.seasons;
     });
   },
   methods: {
     checkAndSelectSeason(seasonId) {
-      const requestedSeason = this.seasons.find(({ id }) => id === seasonId);
-      this.selectedSeason = requestedSeason || this.seasons[0];
+      const requestedSeason = () => { return this.seasons.find(({ id }) => id === seasonId); };
+      if (requestedSeason().rankings.length) {
+        return this.selectedSeason = requestedSeason();
+      }
+
+      io.socket.get(`/stats/seasons/${seasonId}`, ({gameCounts,rankings, uniquePlayersPerWeek}) => {
+        if (!rankings) {
+        this.error = true;
+        return;
+        }
+        
+        this.seasons = [...this.seasons].map((season) => season.id === seasonId ?
+          { ...season, gameCounts, rankings, uniquePlayersPerWeek }
+          : season);
+        this.selectedSeason = requestedSeason();
+      });
```

Comment:
So the api just returns an object `{gameCounts, rankings, uniquePlayersPerWeek}` which is what is missing from each season that gets returned when we call the api for current . so what this does is it loops through the seasons, finds the one that you're requesting to view, and fills up the game counts rankings, and unique players...

this is also why requestedSeason() is a function, so it will refresh after `this.seasons` is updated